sklearn/datasets/openml.py

```diff
@@ -0,0 +1,157 @@
+        Specify another download and cache folder for the data sets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
+
+    target_column : string or None, default 'default-target'
+        Specify the column name in the data to use as target. If
+        'default-target', the standard target column a stored on the server
+        is used. If ``None``, all columns are returned as data and the
+        tharget is ``None``.
+
+    memory : boolean, default=True
+        Whether to store downloaded datasets using joblib.
+
+    Returns
+    -------
+
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
+        'data', the data to learn, 'target', the regression target or
+        classification labels, 'DESCR', the full description of the dataset,
+        'feature_names', the original names of the dataset columns, and
+        'details' which provide more information on the openml meta-data.
+    """
+    data_home = get_data_home(data_home=data_home)
+    data_home = join(data_home, 'openml')
+    if memory:
+        mem = Memory(join(data_home, 'cache'), verbose=0).cache
+    else:
+        def mem(func):
+            return func
+    _get_data_info_by_name_ = mem(_get_data_info_by_name)
+    _get_data_description_by_id_ = mem(_get_data_description_by_id)
+    _download_data_ = mem(_download_data)
+
+    if not exists(data_home):
+        os.makedirs(data_home)
+
+    # check if dataset id is known
+    if isinstance(name_or_id, numbers.Integral):
+        if version != "active":
+            raise ValueError(
+                "Dataset id={} and version={} passed, but you can only "
+                "specify a numeric id or a version, not both.".format(
+                    name_or_id, version))
+        data_id = name_or_id
+    elif isinstance(name_or_id, str):
+        data_info = _get_data_info_by_name_(name_or_id, version)
+        data_id = data_info['did']
+
+    else:
+        raise TypeError(
+            "Invalid name_or_id {}, should be string or integer.".format(
+                name_or_id))
+
+    data_description = _get_data_description_by_id_(data_id)
+    if data_description['status'] != "active":
+        warn("Version {} of dataset {} is inactive, meaning that issues have"
+             " been found in the dataset. Try using a newer version.".format(
+                 data_description['name'], data_description['version']))
+    if target_column == "default-target":
+        target_column = data_description.get('default_target_attribute', None)
+
+    # download actual data
+    data, meta = _download_data_(data_description['url'])
+    columns = np.array(meta.names())
+    data_columns = columns[columns != target_column]
+    # TODO: stacking the content of the structured array
+    # this results in a copy. If the data was homogeneous
+    # we could use a view instead.
+    dtype = object if "nominal" in meta.types() else None
```

Comment:
arff also has a string type (apart from nominal). Is it not more robust to check that everything is numeric and otherwise use object?